demos/smart_framing_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,344 @@
+        if (!FLAGS_labels.empty()) {
+            coco_labels = DetectionModel::loadLabels(FLAGS_labels);
+        }
+
+        /** Configure networks **/
+        cv::gapi::GNetPackage networks;
+        G_API_NET(YOLOv4TinyNet, <GMat2(cv::GMat)>, "yolov4tiny_detector");
+        const auto net = cv::gapi::ie::Params<YOLOv4TinyNet>{
+            FLAGS_m_yolo,                         // path to topology IR
+            fileNameNoExt(FLAGS_m_yolo) + ".bin", // path to weights
+            FLAGS_d_yolo
+        }.cfgOutputLayers({ "conv2d_20/BiasAdd/Add", "conv2d_17/BiasAdd/Add" }).cfgInputLayers({ "image_input" });
+        G_API_NET(SRNet, <cv::GMat(cv::GMat)>, "super_resolution");
+        G_API_NET(SRNet3ch, <cv::GMat(cv::GMat, cv::GMat)>, "super_resolution_3ch");
+        if (apply_sr) {
+            using sr_net = cv::gapi::ie::Params<SRNet>;
+            using sr_net3ch = cv::gapi::ie::Params<SRNet3ch>;
+            auto sr = sr_net{
+                FLAGS_m_sr,                         // path to topology IR
+                fileNameNoExt(FLAGS_m_sr) + ".bin", // path to weights
+                FLAGS_d_sr
+            };
+            auto sr3ch = sr_net3ch{
+                FLAGS_m_sr,                         // path to topology IR
+                fileNameNoExt(FLAGS_m_sr) + ".bin", // path to weights
+                FLAGS_d_sr
+            }.cfgInputLayers({ "1", "0" });
+
+            if (use_single_channel_sr) {
+                networks = cv::gapi::networks(net, sr);
+            } else {
+                networks = cv::gapi::networks(net, sr3ch);
+            }
+        } else {
+            networks = cv::gapi::networks(net);
+        }
+
+        /** ---------------- Main graph of demo ---------------- **/
+        // Now build the graph
+        cv::GMat in;
+        cv::GMat out; //cropped resized output image
+        cv::GMat out_sr; //cropped resized output image after super resolution
+        cv::GMat out_sr_pp; //cropped resized output image after super resolution post processing
+        cv::GMat blob26x26; //float32[1,255,26,26]
+        cv::GMat blob13x13; //float32[1,255,13,13]
+        cv::GArray<custom::DetectedObject> yolo_detections;
+        cv::GArray<std::string> labels;
+
+        std::tie(blob26x26, blob13x13) = cv::gapi::infer<YOLOv4TinyNet>(in);
+        yolo_detections = custom::GYOLOv4TinyPostProcessingKernel::on(in, blob26x26, blob13x13, labels, FLAGS_t_conf_yolo, FLAGS_t_box_iou_yolo, FLAGS_advanced_pp);
+        if (FLAGS_crop_with_borders) {
+            out = custom::GSmartFramingMakeBorderKernel::on(in, yolo_detections);
+        } else {
+            out = custom::GSmartFramingKernel::on(in, yolo_detections);
+        }
+        if (apply_sr) {
+            if (use_single_channel_sr) {
+                cv::GMat b, g, r;
+                std::tie(b, g, r) = cv::gapi::split3(out);
+                auto out_b = custom::GCvt32Fto8U::on(cv::gapi::infer<SRNet>(b));
```

Comment:
That looks strange. Why would anyone train a model that works with separate channels? It's more efficient to have just 3 channel image as input. Are you sure your internal model expects three separate channels, but not grayscale image?